authorizers/actionlist/actionlist.go

```diff
@@ -132,3 +132,92 @@ func validateAction(agent string, action string, claims jwt.MapClaims, log *logr
 	// nothing matched all passed or list is empty, deny all
 	return false, nil
 }
+
+func regoFuncRequiresFactFilter(req *mcorpc.Request) func(_ rego.BuiltinContext, a *ast.Term) (*ast.Term, error) {
+	return func(_ rego.BuiltinContext, a *ast.Term) (*ast.Term, error) {
+		str, ok := a.Value.(ast.String)
+		if !ok {
+			return ast.BooleanTerm(false), fmt.Errorf("invalid fact matcher received")
+		}
+
+		want, err := client.ParseFactFilterString(string(str))
+		if err != nil {
+			return ast.BooleanTerm(false), fmt.Errorf("invalid fact matcher received: %s", err)
+		}
+
+		for _, f := range req.Filter.Fact {
+			if want.Fact == f.Fact && want.Operator == f.Operator && want.Value == f.Value {
+				return ast.BooleanTerm(true), nil
+			}
+		}
+
+		return ast.BooleanTerm(false), nil
+	}
+}
+
+func validateRegoPolicies(req *mcorpc.Request, site string, policies []interface{}, claims jwt.MapClaims, log *logrus.Entry) (ok bool, err error) {
+	for _, policyi := range policies {
+		policy, ok := policyi.(string)
+		if !ok {
+			return false, fmt.Errorf("invalid rego policy claim found in token %v", policyi)
+		}
+
+		data := make(map[string]interface{})
+		err = json.Unmarshal(req.Data, &data)
+		if err != nil {
+			return false, fmt.Errorf("could not parse data embedded in request: %v", err)
+		}
+
+		r := rego.New(
+			rego.Query("data.choria.aaa.policy.allow"),
+			rego.Module("choria.rego", policy),
+			rego.Function1(&rego.Function{Name: "requires_fact_filter", Decl: types.NewFunction(types.Args(types.S), types.B)}, regoFuncRequiresFactFilter(req)),
+			rego.Input(map[string]interface{}{
+				"agent":      req.Agent,
+				"action":     req.Action,
+				"data":       data,
+				"sender":     req.SenderID,
+				"collective": req.Collective,
+				"ttl":        req.TTL,
+				"time":       req.Time,
+				"filter":     req.Filter,
+				"site":       site,
+				"claims":     claims,
+			}),
+		)
```

Comment:
There's some error handling in the docs here about how to spit out things when the compiler is angry at you, so that you can get line numbers and things.